### PR TITLE
change google-conv product name to be based on the base_url

### PR DESCRIPTION
**Pull request by ScottSuarez**
related to https://github.com/GoogleCloudPlatform/terraform-validator/issues/268
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/templates/validator/resource_conversion.go.erb [hunk 1]
```diff
@@ -22,7 +22,8 @@ import (
     resource_name = product_ns + object.name
     properties = object.all_user_properties
     # See discussion on asset name here: https://github.com/GoogleCloudPlatform/magic-modules/pull/1520
-    asset_name_template = '//' + product_ns.downcase + '.googleapis.com/' + (!object.self_link.nil? && !object.self_link.empty? ? object.self_link : object.base_url + '/{{name}}')
+		product_backend_name=@base_url.split('://')[1].split('.googleapis.com')[0]
```
Comment (melinath) [inline mmv1/templates/validator/resource_conversion.go.erb:25]: nit: looks like the indentation here is incorrect

**Review by melinath — COMMENTED** [review]
This looks like it might work - would you be able to confirm that it doesn't break any TFV tests & that it doesn't bring any supported resources out of line with https://cloud.google.com/asset-inventory/docs/supported-asset-types?

File: mmv1/templates/validator/resource_conversion.go.erb (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/templates/validator/resource_conversion.go.erb]: updates

**Comment by ScottSuarez** [comment]
fixes the following validator resources
```
google_filestore_instance    (ok)

google_kms_crypto_key        (ok)
google_kms_key_ring          (ok)

google_sql_database_instance (ok)
```

**Review by melinath — CHANGES_REQUESTED** [review]
💯  This looks great! Looking at the PR again, there are a couple additional spots this fix needs to be applied. Basically, we need to make the same change to the [asset type definition](https://github.com/GoogleCloudPlatform/magic-modules/pull/5203/files#diff-fc27059e45069478256d5b005400b5905d3a9597203ed117a7bcae5a59de68ffR42), and then make the same two changes to the [iam_consumer template](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/templates/validator/iam/iam_consumer.go.erb). Sorry I didn't notice these yesterday but hopefully it'll be a quick change with no unexpected side effects :-)

File: mmv1/templates/validator/resource_conversion.go.erb [hunk 2]
```diff
@@ -38,10 +39,10 @@ func Get<%= resource_name -%>CaiObject(d TerraformResourceData, config *Config)
     if obj, err := Get<%= resource_name -%>ApiObject(d, config); err == nil {
         return []Asset{{
             Name: name,
-            Type: "<%= product_ns.downcase -%>.googleapis.com/<%= object.name -%>",
+            Type: "<%= product_backend_name.downcase -%>.googleapis.com/<%= object.name -%>",
             Resource: &AssetResource{
                 Version: "<%= api_version -%>",
-                DiscoveryDocumentURI: "https://www.googleapis.com/discovery/v1/apis/<%= product_ns.downcase -%>/<%= api_version -%>/rest",
+                DiscoveryDocumentURI: "https://www.googleapis.com/discovery/v1/apis/<%= product_backend_name.downcase -%>/<%= api_version -%>/rest",
```
Comment (melinath) [inline mmv1/templates/validator/resource_conversion.go.erb:45]: Does using the product backend name here as well work / fix an issue?

**Review by melinath — APPROVED** [review]
LGTM. I changed this to "related to" the bug because the PR to update the version of tgc used by TFV is the one that will fix the bug.

File: mmv1/templates/validator/resource_conversion.go.erb (hunk 2 above)
Comment (ScottSuarez) [reply mmv1/templates/validator/resource_conversion.go.erb:45]: yes the previous url doesn't exist if you check the downstreams

**Pull request merged**